### DISCUSSION TITLE
Questions about train, test and validation sets

### DISCUSSION THREAD

**Discussion by kennedyk1** [ultralytics/ultralytics#17276] [General]
Hello everyone!

I have a few questions about the use of train, test, and validation sets during my model training process that remain unclear. I’ll describe my situation below to provide more context:

- I split my dataset into two folders: `train` and `test`.
- In the `rgb.yaml` file, I set the test folder as the `validation` set.

```yaml
path: '/home/user/my_datasets/'
train: 'dataset_1/train/rgb/'
val: 'dataset_1/test/rgb/'
# class names
names:
  0: 'human'
```

- In the main code, I disabled early stopping by setting `patience=0` and trained the model for 50 epochs.
- I used the `low-scratch.yaml` hyperparameter profile to enhance data augmentation.

My main code:
```python
from ultralytics import YOLO
datasets = ['rgb.yaml']
yolo_models = ['yolov6s.yaml','yolov5s.yaml','yolov8s.yaml','yolov9s.yaml','yolov10s.yaml','yolo11s.yaml']
epochs = 50

for dataset in datasets:
    for yolo_model in yolo_models:
        model = YOLO(yolo_model,task='detect') # Create a new model to train
        model.train(
            epochs = epochs, #NUMBER EPOCHS TO TRAIN
            batch = 0.9, #TO USE 90% GPU MEMORY
            single_cls = True, #TO TRAIN WITH ONLY ONE CLASS
            imgsz = 640, # TO REZISE IMAGES, DEFAULT 640
            data = dataset, #TO DEFINE YAML DATASET FILE
            patience = 0, #TO AVOID EARLY STOP
            save = True, #TO SAVE CHECKPOINTS AND FINAL MODEL WEIGHTS
            project = yolo_model.split('.')[0], #NAME OF PROJECT
            name = dataset.split('.')[0], #SUB-NAME OF PROJECT or MODALITY
            plots = True #TO SHOW PLOTS OF TRAINNING AND VALIDATION METRICS
        )
```

Based on the details above, I have the following questions:

-    When is data augmentation applied?
-   Is data augmentation performed on the train set, the validation set, or both?
-    At what stage is validation conducted?
-    Can my test folder be considered "unseen" data, given that I have disabled early stopping?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @kennedyk1, thank you for reaching out about your training setup with Ultralytics 🚀! We understand that navigating through train, test, and validation sets can be a bit complex and we're here to help.

For a better understanding and assistance, it's great that you've provided your setup details. If you encounter any issues or bugs, please make sure to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us diagnose the problem more effectively.

In the meantime, here are a few resources you might find helpful:
- For more details on dataset configuration, check out the [Docs](https://docs.ultralytics.com) which include sections on [Python usage](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/).
- Enhance your model training by following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/), which might clarify some of your questions regarding data augmentation and validation processes.

Join the Ultralytics community wherever you feel most comfortable! For real-time discussions, you can connect with us on [Discord](https://ultralytics.com/discord) 🎧. Prefer a slower pace for more detailed inquiries? Engage with peers on [Discourse](https://community.ultralytics.com) or explore threads and share insights on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade and Setup

Make sure you're using the latest `ultralytics` package by upgrading your environment with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) satisfied:

```bash
pip install -U ultralytics
```

Ensure your environment meets the specifications: [**Python>=3.8**](https://www.python.org/) and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

## Verified Environments

Experience YOLO in various settings with all dependencies ready to go:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: Follow the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: Visit the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: Check the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing, confirming the correct functionality across various OS platforms.

This is an automated response, and an Ultralytics engineer will assist you shortly to delve deeper into your questions and provide tailored guidance. Stay tuned!

**Comment by glenn-jocher** [comment]
@kennedyk1 data augmentation is applied only to the training set. Validation is conducted after each epoch to evaluate model performance. Your test folder can be considered "unseen" data if it's not used during training or validation. For more details, visit the [model testing guide](https://docs.ultralytics.com/guides/model-testing/).
